toolkits/confluence/arcade_confluence/client.py

```diff
@@ -0,0 +1,535 @@
+        self.api_version = api_version.value
+
+    def _get_cloud_id(self) -> str:
+        """
+        Fetch the cloudId for <workspace_name>.atlassian.net
+        using the OAuth2 3LO accessible-resources endpoint.
+
+        For details on why this is necessary, see: https://developer.atlassian.com/cloud/oauth/getting-started/making-calls-to-api
+        """
+        headers = {"Authorization": f"Bearer {self.token}"}
+        resp = httpx.get(self.ACCESSIBLE_RESOURCES_URL, headers=headers)
+        resp.raise_for_status()
+        resp_json = resp.json()
+
+        if len(resp_json) == 0:
+            raise ToolExecutionError(message="No workspaces found for the authenticated user.")
+
+        return resp_json[0].get("id")  # type: ignore[no-any-return]
+
+    async def request(self, method: str, path: str, **kwargs: Any) -> Any:
+        headers = {
+            "Accept": "application/json",
+            "Authorization": f"Bearer {self.token}",
+        }
+        async with httpx.AsyncClient() as client:
+            response = await client.request(
+                method,
+                f"{self.BASE_URL}/{self.cloud_id}/{self.api_version}/{path.lstrip('/')}",
+                headers=headers,
+                **kwargs,
+            )
+            response.raise_for_status()
+            return response.json()
+
+    async def get(self, path: str, **kwargs: Any) -> Any:
+        return await self.request("GET", path, **kwargs)
+
+    async def post(self, path: str, **kwargs: Any) -> Any:
+        return await self.request("POST", path, **kwargs)
+
+    async def put(self, path: str, **kwargs: Any) -> Any:
+        return await self.request("PUT", path, **kwargs)
+
+
+class ConfluenceClientV1(ConfluenceClient):
+    def __init__(self, token: str):
+        super().__init__(token, api_version=ConfluenceAPIVersion.V1)
+
+    def construct_cql(
+        self, terms: list[str] | None, phrases: list[str] | None, enable_fuzzy: bool = False
+    ) -> str:
+        """Construct a CQL query from a list of terms and phrases.
+
+        Args:
+            terms: list of single words to search for
+            phrases: list of multiple words to search for
+            enable_fuzzy: enable fuzzy matching to find similar terms (e.g. 'roam' will find 'foam')
+
+        Learn about advanced searching using CQL here: https://developer.atlassian.com/cloud/confluence/advanced-searching-using-cql/
+        """
+        cql_parts = []
+
+        # Process single terms
+        if terms:
+            terms = split_by_space(terms)
+            term_queries = []
+            for term in terms:
+                term_suffix = "~" if enable_fuzzy else ""
+                term_queries.append(
+                    f'(text ~ "{term}{term_suffix}" OR title ~ "{term}{term_suffix}" OR space.title ~ "{term}{term_suffix}")'  # noqa: E501
+                )
+
+            if term_queries:
+                cql_parts.append(f"({' OR '.join(term_queries)})")
```

Comment:
While testing the tool, I realized it's OR'ing each term and phrase provided. My intuition is that if I search using `["hello", "world"]`, I'd like to see documents that contain both words.

I think using OR to concatenate will yield poor search results, a low signal to noise ratio. My preference would be to AND all terms and phrases.